pkg/querier/querier.go

```diff
@@ -377,6 +379,120 @@ func (q *Querier) tailDisconnectedIngesters(ctx context.Context, req *logproto.T
 	return reconnectClientsMap, nil
 }
 
+// Series fetches any matching series for a list of matcher sets
+func (q *Querier) Series(ctx context.Context, req *logproto.SeriesRequest) (*logproto.SeriesResponse, error) {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = q.validateQueryTimeRange(userID, &req.Start, &req.End); err != nil {
+		return nil, err
+	}
+
+	// Enforce the query timeout while querying backends
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(q.cfg.QueryTimeout))
+	defer cancel()
+
+	series, errs := make(chan [][]logproto.SeriesIdentifier, 2), make(chan error, 2)
```

Comment:
Personally, I find initializing two values of two different types on the same line a little hard to read; can you split them out?